### ISSUE TITLE
Some elasticsearch-ci/docs CI job never report

### ISSUE THREAD

**Issue by tylersmalley** [elastic/docs#2071]
There have been a handful of CI jobs over the past couple of days that have failed to report their status. Here is a sample:

https://github.com/elastic/kibana/pull/91408
https://github.com/elastic/kibana/pull/91914

Since they are never reported or linked back to a job, I have been unable to find the logs in the job to see what happened. 

Is this something that is known?

**Comment by gtback** [comment]
@tylersmalley I've seen it happen before, but not in the Kibana repo. In the past, I think it's happened in newer repos where Jenkins doesn't have permission to update the GitHub Check status. But that doesn't seem to be the case here. I'll look into the two you mention above.

**Comment by gtback** [comment]
It looks like those jobs are too old and may have been deleted. I'll start another one for the PR that's still open and see if I can tell anything.

**Comment by gtback** [comment]
It seems like the [build I just kicked off](https://elasticsearch-ci.elastic.co/view/Docs/job/elastic+kibana+pull-request+build-docs/3599/) reported its status correctly to the PR, and the "details" link contained the correct URL to the job. We might just have to wait until this happens again and I can dig in more.

**Comment by gtback** [comment]
cc: @elastic/infra-automation . Have you fine folks ever seen anything like this before?

**Comment by stratoula** [comment]
Hey guys this happens to all the PRs created by our collaborators and in order to unblock them, I run every time `@elasticmachine run elasticsearch-ci/docs` on their PRs. Moreover, it happens to the backport PRs too so I would really appreciate it if we could solve it 😄 🙏

**Comment by gtback** [comment]
@stratoula do you have a link to any PRs (or if you have them, the jobs themselves) so I can look into the logs and see what's going on?

**Comment by stratoula** [comment]
Yes so I have this one https://github.com/elastic/kibana/pull/91293

And these are some PRs that they had the problem and when I run the above command it was solved:
https://github.com/elastic/kibana/pull/92183
https://github.com/elastic/kibana/pull/92170
https://github.com/elastic/kibana/pull/91838

**Comment by gtback** [comment]
Ah, thanks @stratoula, this actually makes sense to me now. For good reason, by default we don't run CI automatically on PRs submitted by people outside the Elastic organization. In this case they are "trusted" to contribute to Kibana even though they aren't members of the organization, so all of the CI jobs run, except for the docs ones which run on elasticsearch-ci rather than kibana-ci.

I'll look into the best way we can fix it, but for now manually adding that comment is the best workaround.

**Comment by stratoula** [comment]
Yes I would really appreciate it if we can fix it somehow because our external contractors are highly dependent on an Elastic member in order for their PRs to be merged and as it also happens for the backport PRs, it may affect our productivity. cc @timroes

**Comment by gtback** [comment]
Thanks, I'll prioritize getting a fix for this.

**Comment by itsmed** [comment]
@gtback are these collaborators a part of a GitHub team? If so, then I believe all that needs to be done is to give that team `write` access to the repo(s) requested.  If they are not a part of a GitHub team, we should add them to a team. We should also allow @stratoula or whomever supervises these folks to be an administrator of the team so that we don't slow folks down in the future.

**Comment by timroes** [comment]
@itsmed Since we use Okta as single sign on on Github, we can't add non Okta users to teams anymore, thus it's unfortunately impossible for us to add those people into a team under the `elastic` org on GitHub. Please let me know if there is any way around this, but I tried this for teams I am admin on and I don't have a chance of adding them there.

Regarding the trust issue and outside of Elastic: There are still different level of "assosciation" in GitHub. I don't see why we couldn't add "Collaborator" to have that run automaticalley. A collaboratir is everyone we explicitly gave write permissions to some repository. Not to confuse with "Contributor" which is everyone who have in the past contributed to one of the repositories. Imho the easiest fix for this would be to allow docs to run automatically for Collaborators. Is there any specific concern against having it run automatically for people we anyway trusted with write permission to the repository?

**Comment by itsmed** [comment]
> @itsmed Since we use Okta as single sign on on Github, we can't add non Okta users to teams anymore, thus it's unfortunately impossible for us to add those people into a team under the elastic org on GitHub. Please let me know if there is any way around this, but I tried this for teams I am admin on and I don't have a chance of adding them there.

Ahhh, ok the missing step! @timroes we should be able to get an Okta group set up for the collaborators by involving @elastic/it-operations-eng-admins I think @rickpollak or @Franny2times would be able to set up the Okta group for you as has been done for a couple of other similar requests, though they may request an email to `helpdesk@`.

>  Regarding the trust issue and outside of Elastic: There are still different level of "assosciation" in GitHub. I don't see why we couldn't add "Collaborator" to have that run automaticalley. 

I know that we locked down access as part of going public for auditing and compliance purposes, so I don't think that we can give access freely to folks outside of the organization; with that said let's ping @elasticdog to confirm.
